coverage scores + get filtered samples

clrnb opened the pull request:
coverage scores -> plot cluster and class coverage scores for each layer of nn
get filtered samples -> util function to get index of samples based on class name, correct/incorrect prediction, confidence range,...

Commit:
initial commit

Commit:
minor

Commit:
update readme

lorenzoCapelli committed:
pop norm

lorenzoCapelli committed:
Merge branch 'norm'

lsrosa committed:
minor for atks (#1)

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lsrosa committed:
cleanup (#4)

clrnb committed:
Allow custom transformations for CIFAR dataset  (#3)

allows custom compose transformation and augmentation
* set_model allows state_dict stored directly

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

saravorabbi committed:
add check for ViT model (#7)

lorenzoCapelli committed:
both binary and multilabel (#12)

* both binary and multilabel

* parser_pred->pred_fn

lorenzoCapelli committed:
11 empp labels (#13)

* ep label free

* still getting through

* diagnosis works we have to deal with Nan

* c'è problema con i NaN

* minor ciao

* filter_nan removed

* minor

---------

Co-authored-by: Leandro <le.souza.rosa@gmail.com>

lsrosa committed:
Samplers (#17)

* test

* add release notes

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
5 dependencies (#19)

* first

* add dependencies

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

clrnb committed:
Clarabranch (#14)

* wip

no bias svd (cleaned version)

group handeling (wip)

group handeling working (error when bias=true)

minor

svd w/group completed

update get_svd interface

verbose

clean import

* minor cleanup

* update release noters

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
9 peephole multi layer support (#20)

* update Peephole obj fields + update multi-layers handling in get_peepholes()

get_scores() and evaluate_dists() update for multi-layers handling

* wip

* add multiple layers support

* update release notes

* clean imports

---------

Co-authored-by: saravorabbi <sara.vorabbi@gmail.com>
Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
6 refactor get corevectors (#22)

* first attempt

* ahhhhh

* still coding

* making some order

* we just miss a dummy reduction

* add in_actv parser

minor cleanups

* change size to shape

* major che mi ero scordato

* norm prob

* minor

* fix normalization

* remove dataloader from dataset

* fix bugs

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>
Co-authored-by: Leandro <le.souza.rosa@gmail.com>

lorenzoCapelli committed:
15 save peepholes classifiers parameters and configuration (#23)

* modifing the classifier_base
save classifiers

* load and save classifiers

* minor cleanup

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
loading problems solved (#27)

Neat!

lorenzoCapelli committed:
24 refactor get peepholes (#26)

* minor

* non so se va bene ma arriva in fondo a ph

* minor save

* adversaries

* mi manca il backprob

* still working

* done with the call

* still working

* ready for PR

* reorganize

* minor fix

* minor

* clean fit()

* WIP

* clean DMD __call__()

(WIP) solve mem allocation

* (WIP) debug memory

* stabilize mem

* fix bs

* it is a bit slow

* test done

* polish tqdm

* update Release notes

* update Release notes

* update Release notes

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
minor (#29)

target modules

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lsrosa committed:
update saving drillers (#32)

* update saving drillers

* fix layer to module on conv svd

* fix bug batch size bug

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

clrnb committed:
fix module-layers (#33)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
abbiamo una funzione (#35)

lsrosa committed:
add ch_wise svd to get_svd() (#37)

get cv size from data

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

clrnb committed:
Viz (#36)

* add todo comments

* add fn

* conceptogram functions (useless :P)

---------

Co-authored-by: saravorabbi <sara.vorabbi@gmail.com>
Co-authored-by: Leandro <le.souza.rosa@gmail.com>

lsrosa committed:
:wqmove fns out of peepholes (#39)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
add cols (#41)

bluelivi committed:
ImageNet-1k data loading (#48)

* ImageNet default transforms + augment policy

* imagenet train and val loader mirroring the cifar data loader

* fixed typo

lorenzoCapelli committed:
46 feature squeezing implementation (#49)

* featuresqueezing starting

* almost done with feature squeezing

* minor misspelling

lorenzoCapelli committed:
adjusting some studff (#51)

lsrosa committed:
Add function to fine tune models (#47)

* Add function to fine tune models

Small cleanups on corvectors

* rework

* rework saving, loading, and iter

* minor

* clean imagenet

* document better

* fix bugs. add accuracy plots

* add scheduler

* save and load scheduler

* add check for LRO scheduler

* test

* add data parallel

* remove print

* better print

* debug acc, improve checking

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
plot conceptograms (#53)

* plot conceptograms

* remove bak

* minor

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>
Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lorenzoCapelli committed:
44 activations saving (#52)

Remove the necessity of saving activations.
Rework Hooks, run `dry_runs` internally, so it is not need on experiments

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>
Co-authored-by: Leandro de Souza Rosa <lsrosa@users.noreply.github.com>

lsrosa committed:
update DMD. update conceptogram (#54)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
dry run on svds (#55)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
56 making attacks compatible with datasetbase (#57)

* see if it works

* getting this get

* it seems ok

* I was forgetting imagenet

* moving unsqueeze to .get()

* is it too late to say sorry

* sembra anda

* tolto if messo unsqueeze

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
minor (#58)

lsrosa committed:
Implemente convolution with a line score (#61)

solve SVD memory blobbing

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
i think it is enough (#62)

Add `mode` as kwargs for `load_only` fns

lorenzoCapelli committed:
minor not so minor (#63)

clrnb committed:
solve conflicts

minor

score_type added (max)

viz better for pr

average coverage plot

conv viz singular values

a lil fix

visualize corevectors (raw and with t-sne)

random projection implementation

small changes viz corevector + imagenet transf

small changes

bug fix

clrnb reviewed (commented):
(also minor stuff to make conceptogram prettier in superconceptogram

lsrosa committed:
Implement Kernel SVD (#65)

* implement kernel svd

fix small bug on conceptograms. Update documentation a bit

fix svd devices

* update docs

* remove unused return on unroll_image

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

bluelivi committed:
Evaluate auc (#64)

* conceptogram_cl_score now returns metrics dict with auc

* top-k accuracy computation function added

* replace np and sklearn AUC with torcheval.metrics.AUC per PR feedback

lsrosa reviewed (changes requested):
The code needs polishing. Also, rebase the branch onto main.

clrnb commented on peepholelib/utils/viz_empp.py:
```diff
@@ -0,0 +1,165 @@
+    """
+    class_scores = get_empp_class_coverage_scores(drillers, threshold)
+    cluster_scores = get_empp_cluster_coverage_scores(drillers, threshold)
+
```
They're also used in plot_empp_scores

lsrosa committed:
Fix AUC functions (#67)

* fix AUC

* update kmeans

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
use torch.svd_lowrank (#68)

add use_s to projection
remove prints

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

clrnb commented on peepholelib/utils/mappings.py:
```diff
@@ -14,3 +14,26 @@ def map_labels(**kwargs):
                 mapped_labels.append(new_label)
     return torch.tensor(mapped_labels)
+
+coarse_to_fine_cifar100 = {
```
you still need the superclass names somehow so I will do map superclass name -> indeces of fine class

lsrosa committed:
Implement protoclass scores (#71)

solve bugs on scores

cosmetic touches

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
update plots, add docs (#72)

lsrosa committed:
minor (#70)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
add vgg16 vit transform (#76)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
add dropping accuracy (#77)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
Add max_drop to plots (#78)

* add max drop

* reflect on entropy

* minor polish

* move data to main mem before GMM

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
changed score computation (#79)

lorenzoCapelli committed:
Concepto net (#80)

* just setting up stuff

* done with this

lsrosa committed:
Add Cifar 100 C (#81)

* working on visualizing something

something wrong with concepto

doing some experiments

WIP

minor cleanup

readd entropy

oops

WIP add corrupted

polish

add loaders to cvs norm. Add some doc

WIP images are broken

fix transpose

fix path

calib

minor

separate corruption intensity

touches

* solve comments

---------

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>
Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
Update empirical posterior interface (#83)

* Update empirical posterior interface

* update doc

---------

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lsrosa committed:
rework scores and plots (#84)

* rework scores and plots

* the label error again

---------

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lsrosa committed:
add odd dist plots (#85)

* add odd dist plots

* add missing check

---------

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lsrosa committed:
fix small bug. add loaders to get_peepholes (#86)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
add loaders to plot calibration (#87)

* add loaders to plot calibration

* WIP plots

* polish plots

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
VAL for OOD (#88)

* spero basti

* tried to fix it

lorenzoCapelli committed:
non mi funzionava (#89)

lsrosa committed:
Many changes Corevectors vs Attacks (#91)

implement RelU

add hdbscan test

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lsrosa committed:
Add other OOD datasets (#93)

* Add SVHN, Places, and DTD as OOD samples to CIFAR

* save ood in the same order as ds

* minor reviews

---------

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>
Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
fix when no ood is passed (#94)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
Solves concepto visualization but. Add DMD++. (#100)

* makes nsamples close to CIFAR

make target modules mandatory

show both MD and MDplus

change calibration to line plot

* minor unmaking

---------

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lorenzoCapelli committed:
Introducing DOCTOR DMD++ scores (#103)

* make nsamples close to CIFAR

get score dmd basic

touches on plots

make target modules mandatory

show both MD and MDplus

update attacks wrt to ood

implements doctor

* revert cifar to main

* revert atk base

* minor polish

* minor polish on doctor

* fpr func

* clean

---------

Co-authored-by: Leandro <le.souza.rosa@gmail.com>
Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
Rework datasets and atcks (#105)

reflect on cvs, phs, scores, plots

separate datasetWrap and parsedDS

Co-authored-by: lorenzo <lorenzo.capelli@outlook.com>

lsrosa committed:
Fixes after testing (#110)

* Update atks, fixes on scores and plots

* add title arguments

* polish plots

---------

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
fit gmm until convergence (#112)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
select only successfull atk samples (#113)

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
Fix bug on DMD (#114)

clrnb reviewed (commented):
what is Branch protection policy

lorenzoCapelli committed:
Add input normalizer

implement function that adds a normalization layer to the beginning of the model.

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lsrosa committed:
Implement MD, NLL, cv GMM score. (#118)

Del version of Fiorire

Co-authored-by: Leandro <leandro.desouzarosa@unibo.it>

lorenzoCapelli committed:
Lollo again (#116)

* primo confidence score for dmd

* minor change in colormap of conceptograms

* sto quasi in fondo ma i risultati non sono incoraggianti

* i think there is something different

* non so che ci sono

* just viz stuff

* removing some stuff before pushin

* almost done

* removed some leftovers

* going on and on

* i tried to modify the stuff you asked me for
